plugin/backend/ipvariant.d

```diff
@@ -678,43 +675,66 @@ body {
 }
 
 
-@trusted void generateClass(Generator.Modules inner, string[] ns, string type, SUTEnv sut) {
+@trusted CppModule generateClass(Generator.Modules inner, string class_name, string[] ns, string type, SUTEnv sut) {
+    //Some assumptions are made. Does all interfaces start with I_? Does all providers and requirers end with Requirer or Provider?
```

Comment:
good times, writing down the assumptions :)
next step, if possible, is to encode them in Contracts.